pkg/gateway.go

```diff
@@ -68,6 +70,17 @@ func NewGateway(configProvider *config.Provider) (*Gateway, error) {
 
 // Run starts and runs the gateway according to given configuration
 func (gw *Gateway) Run(ctx context.Context) error {
+	// Instrument the gateway process
+	err := host.Start()
```

Comment:
Do we need to call some Stop() method as well? I'm wondering how this would work on service shutdown

Reply:
> Do we need to call some Stop() method as well? 

nope!

otel api is callback based, basically the `host.Start` call registers  callback on the global otel providers.